src/viam/sdk/tests/test_common.cpp

```diff
@@ -0,0 +1,96 @@
+#define BOOST_TEST_MODULE test module test_common
+
+#include <boost/test/included/unit_test.hpp>
+
+#include <viam/api/common/v1/common.pb.h>
+
+#include <viam/sdk/common/utils.hpp>
+
+namespace viam {
+namespace sdktests {
+
+using namespace viam::sdk;
+
+// For common/util.* protobuf duration from/to_proto
+// conversion to ensure we are within the proto
+// spec
+BOOST_AUTO_TEST_SUITE(test_time_conversion)
+
+using Duration = google::protobuf::Duration;
+
+BOOST_AUTO_TEST_CASE(test_zero) {
+    Duration input;
+    input.set_nanos(0);
+    input.set_seconds(0);
+    auto converted = from_proto(input);
+    BOOST_CHECK_EQUAL(converted.count(), 0);
+    auto reconverted = to_proto(converted);
+    BOOST_CHECK_EQUAL(reconverted.nanos(), 0);
+    BOOST_CHECK_EQUAL(reconverted.seconds(), 0);
+}
```

Comment:
This test logic could be refactored into a shared helper function but then the error message points to the wrong line and it is a bit of a pain to understand.